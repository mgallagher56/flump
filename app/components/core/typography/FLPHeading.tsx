import type { PropsWithChildren } from 'react';
import React from 'react';

import type { HeadingProps } from '@chakra-ui/react';
import { Heading } from '@chakra-ui/react';

const FLPHeading: React.FC<PropsWithChildren<HeadingProps>> = (props) => {
  return <Heading {...props}>{props.children}</Heading>;
};

export default FLPHeading;
