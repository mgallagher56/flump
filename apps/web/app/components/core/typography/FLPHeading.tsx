import type { FC, PropsWithChildren } from 'react';

import type { HeadingProps } from '@chakra-ui/react';
import { Heading } from '@chakra-ui/react';

const FLPHeading: FC<PropsWithChildren<HeadingProps>> = ({ color = 'blue.500', ...props }) => {
  return (
    <Heading color={color} {...props}>
      {props.children}
    </Heading>
  );
};

export default FLPHeading;
