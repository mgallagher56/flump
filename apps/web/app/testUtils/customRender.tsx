// create a custom render function with ChakraProvider
import { type ReactNode } from 'react';

import { type RenderOptions, type RenderResult, render as rtlRender } from '@testing-library/react';
import userEvent, { type UserEvent } from '@testing-library/user-event';
import { Provider } from '~/components/ui/provider';

const user = userEvent.setup();

const customRender = (ui: ReactNode, options?: RenderOptions): RenderResult & { user: UserEvent } => {
  const wrappedUi = <Provider>{ui}</Provider>;

  return {
    ...rtlRender(wrappedUi, options),
    user
  };
};

export default customRender;
