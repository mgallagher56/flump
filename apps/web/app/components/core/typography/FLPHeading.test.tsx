import customRender from '~/testUtils/customRender';


import FLPHeading from './FLPHeading';

const sampleText =
  'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nemo quaerat repellendus numquam et esse ut labore molestias nihil optio expedita atque, quas eligendi corrupti quasi impedit suscipit sunt.';

describe('FLPHeading', () => {
  test('renders as expected', () => {
    const { container } =customRender(<FLPHeading as="h1">{sampleText}</FLPHeading>);
    expect(container).toMatchSnapshot();
  });
  test('renders as expected with a size', () => {
    const { container } =customRender(<FLPHeading size="lg">{sampleText}</FLPHeading>);
    expect(container).toMatchSnapshot();
  });
  test('renders as expected with a color', () => {
    const { container } =customRender(<FLPHeading colorPalette="blue">{sampleText}</FLPHeading>);
    expect(container).toMatchSnapshot();
  });
  test('render only number of lines', () => {
    const { container } =customRender(
      <FLPHeading as="h1" size="4xl">
        {sampleText}
      </FLPHeading>
    );
    expect(container).toMatchSnapshot();
  });
});
