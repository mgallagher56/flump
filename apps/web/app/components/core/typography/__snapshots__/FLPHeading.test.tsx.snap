// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FLPHeading > render only number of lines 1`] = `
<div>
  <script
    nonce=""
  >
    ((e,r,s,u,d,m,l,h)=&gt;
    {let c=document.documentElement,v=["light","dark"];function p(i){(Array.isArray(e)?e:[e]).forEach(y=&gt;
    {let k=y==="class",S=k&&m?d.map(f=&gt;
    m[f]||f):d;k?(c.classList.remove(...S),c.classList.add(i)):c.setAttribute(y,i)}),R(i)}function R(i){h&&v.includes(i)&&(c.style.colorScheme=i)}function a(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let i=localStorage.getItem(r)||s,y=l&&i==="system"?a():i;p(y)}catch(i){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <h1
    class="chakra-heading css-155geyk"
  >
    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nemo quaerat repellendus numquam et esse ut labore molestias nihil optio expedita atque, quas eligendi corrupti quasi impedit suscipit sunt.
  </h1>
</div>
`;

exports[`FLPHeading > renders as expected 1`] = `
<div>
  <script
    nonce=""
  >
    ((e,r,s,u,d,m,l,h)=&gt;
    {let c=document.documentElement,v=["light","dark"];function p(i){(Array.isArray(e)?e:[e]).forEach(y=&gt;
    {let k=y==="class",S=k&&m?d.map(f=&gt;
    m[f]||f):d;k?(c.classList.remove(...S),c.classList.add(i)):c.setAttribute(y,i)}),R(i)}function R(i){h&&v.includes(i)&&(c.style.colorScheme=i)}function a(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let i=localStorage.getItem(r)||s,y=l&&i==="system"?a():i;p(y)}catch(i){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <h1
    class="chakra-heading css-mytsjf"
  >
    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nemo quaerat repellendus numquam et esse ut labore molestias nihil optio expedita atque, quas eligendi corrupti quasi impedit suscipit sunt.
  </h1>
</div>
`;

exports[`FLPHeading > renders as expected with a color 1`] = `
<div>
  <script
    nonce=""
  >
    ((e,r,s,u,d,m,l,h)=&gt;
    {let c=document.documentElement,v=["light","dark"];function p(i){(Array.isArray(e)?e:[e]).forEach(y=&gt;
    {let k=y==="class",S=k&&m?d.map(f=&gt;
    m[f]||f):d;k?(c.classList.remove(...S),c.classList.add(i)):c.setAttribute(y,i)}),R(i)}function R(i){h&&v.includes(i)&&(c.style.colorScheme=i)}function a(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let i=localStorage.getItem(r)||s,y=l&&i==="system"?a():i;p(y)}catch(i){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <h2
    class="chakra-heading css-1dct4j5"
  >
    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nemo quaerat repellendus numquam et esse ut labore molestias nihil optio expedita atque, quas eligendi corrupti quasi impedit suscipit sunt.
  </h2>
</div>
`;

exports[`FLPHeading > renders as expected with a size 1`] = `
<div>
  <script
    nonce=""
  >
    ((e,r,s,u,d,m,l,h)=&gt;
    {let c=document.documentElement,v=["light","dark"];function p(i){(Array.isArray(e)?e:[e]).forEach(y=&gt;
    {let k=y==="class",S=k&&m?d.map(f=&gt;
    m[f]||f):d;k?(c.classList.remove(...S),c.classList.add(i)):c.setAttribute(y,i)}),R(i)}function R(i){h&&v.includes(i)&&(c.style.colorScheme=i)}function a(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let i=localStorage.getItem(r)||s,y=l&&i==="system"?a():i;p(y)}catch(i){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <h2
    class="chakra-heading css-e8b6d6"
  >
    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nemo quaerat repellendus numquam et esse ut labore molestias nihil optio expedita atque, quas eligendi corrupti quasi impedit suscipit sunt.
  </h2>
</div>
`;
